ingestion_tools/scripts/importers/alignment.py

```diff
@@ -0,0 +1,180 @@
+        self.identifier = AlignmentIdentifierHelper.get_identifier(config, metadata, parents)
+
+    def get_dest_filename(self) -> str:
+        if not self.path:
+            return None
+        output_dir = self.get_output_path()
+        return f"{output_dir}{os.path.basename(self.path)}"
+
+    def import_metadata(self) -> None:
+        try:
+            meta = AlignmentMetadata(self.config.fs, self.get_deposition().name, self.get_base_metadata())
+            meta.write_metadata(self.get_metadata_path(), self.get_extra_metadata())
+        except IOError:
+            print("Skipping creating metadata for default alignment with no source tomogram")
+
+    def import_item(self) -> None:
+        if self.is_default_alignment():
+            print(f"Skipping importing alignment with path {self.path} as it is a default alignment")
+            return
+        super().import_item()
+
+    def get_output_path(self) -> str:
+        output_directory = super().get_output_path()
+        return os.path.join(output_directory, f"{self.identifier}-")
+
+    def get_metadata_path(self) -> str:
+        return self.get_output_path() + "alignment_metadata.json"
+
+    def get_extra_metadata(self) -> dict:
+        tlt_importer, tltx_importer = self.get_tlt_importers()
+        return {
+            "volume_dimension": self.get_tomogram_volume_dimension(),
+            "per_section_alignment_parameters": self.get_per_section_alignment_parameters(tlt_importer, tltx_importer),
+            "alignment_path": self.get_dest_filename(),
+            "tilt_path": tlt_importer.get_dest_filename() if tlt_importer else None,
+            "tiltx_path": tltx_importer.get_dest_filename() if tltx_importer else None,
+        }
+
+    def get_tomogram_volume_dimension(self) -> dict:
+        for tomogram in TomogramImporter.finder(self.config, **self.parents):
+            return tomogram.get_source_volume_info().get_dimensions()
+
+        # If no source tomogram is found don't create a default alignment metadata file. But, if alignment files are
+        # available, create metadata file with default volume dimension
+        if self.is_default_alignment():
+            raise IOError("No source tomogram found for creating default alignment")
+
+        print("No source tomogram found for alignment, setting volume dimension to default")
+        return {"x": None, "y": None, "z": None}
+
+    def get_per_section_alignment_parameters(
+        self,
+        tlt_importer: BaseFileImporter,
+        tltx_importer: BaseFileImporter,
+    ) -> list:
+        result = []
+        if self.is_default_alignment():
+            return result
+        xf_data = self.get_xf_data()
+        tlt_data = self.get_dataframe(tlt_importer, ["tilt_angle"])
+        tltx_data = self.get_dataframe(tltx_importer, ["volume_x_rotation"])
+        for index, entry in xf_data.iterrows():
+            item = {
+                "z_index": index,
```

Comment:
Are you missing `beam_tilt`?

Reply:
> Are you missing `beam_tilt`?

NVM, I see NVNM I see https://github.com/chanzuckerberg/cryoet-data-portal/issues/898#issuecomment-2329765964